debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java

```diff
@@ -318,8 +318,15 @@ private void createDataEventsForTable(ChangeEventSourceContext sourceContext, Re
         try (Statement statement = readTableStatement();
                 ResultSet rs = statement.executeQuery(selectStatement.get())) {
 
-            Column[] columns = getColumnsForResultSet(table, rs);
-            final int numColumns = table.columns().size();
+            ResultSetMetaData metaData = rs.getMetaData();
+            Column[] columns = new Column[metaData.getColumnCount()];
+            int greatestColumnPosition = 0;
+            for (int i = 0; i < columns.length; i++) {
+                columns[i] = table.columnWithName(metaData.getColumnName(i + 1));
+                greatestColumnPosition = greatestColumnPosition < columns[i].position()
```

Comment:
So in which case would `greatestColumnPosition` be different from the width of the result set (ignoring any +/-1 difference due to indexing based on 0 vs. 1)?

Reply:
They should be the same now, I mentioned in a previous comment:

> I can remove what I did with the column position indexing and go back to raw indexing now, but I think using position still is probably a safer approach, but I can revert that if desired.

Reply:
I feel position is a safer way to go, as I've been implementing several changes around this area, and each time I run into issues around these indexes. Using the position everywhere ensures we have a consistent column size and placement regardless of what happens (columns being excluded and what not).

Reply:
Actually, this might still be necessary now that I think about it.
With the snapshot now filtering items out of the select based on the include/exclude list, if you filter out a column in the middle of the row, it'll run into issues with the raw index...

Reply:
Ok, I suppose you're right. Overall I still have the feeling (or hope?) we could get rid of the position business altogether, but that'll have to wait for another time. This first loop is invoked only once per result set, so overhead is small.